test/library/input_numeric_format.spec.js

```diff
@@ -381,6 +381,22 @@ describe('Test NumberFormat as input with numeric format options', () => {
     expect(wrapper.state().value).toEqual('$0.25');
   });
 
+  it('should not remove leading 0s while user go out of focus and allowLeadingZeros is true', () => {
+    const wrapper = shallow(<NumberFormat value={23456.78} thousandSeparator={','} decimalSeparator={'.'} prefix={'$'} allowLeadingZeros={true}/> );
+
+    simulateKeyInput(wrapper.find('input'), '0', 1);
+    simulateBlurEvent(wrapper.find('input'));
+
+    expect(wrapper.state().value).toEqual('$023,456.78');
+
+    wrapper.setProps({value: 10000.25});
+    wrapper.update();
+
+    simulateKeyInput(wrapper.find('input'), 'Backspace', 2);
+    simulateBlurEvent(wrapper.find('input'));
+    expect(wrapper.state().value).toEqual('$0,000.25');
```

Comment:
It's a wrong test. The leading 0 should be having a thousand separators. I remember fixing this issue. Check after new code pull if this persists?

Reply:
When will the new feature to allow leading zeros be available?

Reply:
I am also be interested in this feature

Reply:
Hi @s-yadav.  I forked this project to complete this feature.  With latest the above test is still passing.  What result are you expecting?  are you expecting '$0000.25' instead of '$0,000.25'

Reply:
Yes. It should be $0000.25